### PR TITLE
[Scene] Use NULL instead of COND checks when appropriate

### PR DESCRIPTION
**Pull request by AThousandShips**
Restricted to scene (it was enough work there, will expand in a later PR)

Tried to get everything but might have missed some

### CODE REVIEW

File: scene/main/viewport.cpp [hunk 1]
```diff
@@ -2339,7 +2339,7 @@ void Viewport::_gui_force_drag(Control *p_base, const Variant &p_data, Control *
 
 void Viewport::_gui_set_drag_preview(Control *p_base, Control *p_control) {
 	ERR_FAIL_NULL(p_control);
-	ERR_FAIL_COND(!Object::cast_to<Control>((Object *)p_control));
+	ERR_FAIL_NULL(Object::cast_to<Control>((Object *)p_control));
```
Comment (AThousandShips) [inline scene/main/viewport.cpp:2342]: This line is technically redundant, but left it in, can remove it if desired, as the pointer is already `Control` this check should be redundant

**Review by KoBeWi — APPROVED** [review]
Looks fine.
NULL is better than COND, because it provides better default message. It doesn't really matter when used with MSG.

**Comment by AThousandShips** [comment]
I agree, but I'd argue that when the context isn't obvious (though most cases here are, as they were the easiest to just search/replace) I find that the type of the check helps with readability of the code, when it's a member variable

File: scene/main/viewport.cpp (hunk 1 above)
Comment (AThousandShips) [reply scene/main/viewport.cpp]: Didn't dig into the blame but the only way this makes sense would be if the argument used to be a `Variant`

**Comment by akien-mga** [comment]
Thanks!

**Comment by AThousandShips** [comment]
Thank you!

**Pull request merged**